File: server/PQL-Domain-Submission/src/main/java/com/passql/submission/dto/ProgressResponse.java
package com.passql.submission.dto;

/**
 * 홈 화면용 사용자 진도 응답.
 *
 * 기존 3필드(solvedCount, correctRate, streakDays)는 다른 화면에서 계속 사용되므로 100% 보존한다.
 * `readiness` 블록은 무한 문제 환경에서 "합격 준비도" 개념으로 산출된 게이지 값이다.
 */
public record ProgressResponse(
    long solvedCount,
    double correctRate,
    int streakDays,
    ReadinessResponse readiness
) {}

File: lib/data/models/progress/progress_response.dart
import 'package:freezed_annotation/freezed_annotation.dart';
import 'readiness_response.dart';

part 'progress_response.freezed.dart';
part 'progress_response.g.dart';

/// GET /progress 응답.
/// correctRate: 0.0~1.0 (마지막 시도 기준)
/// streakDays: 연속 학습 일수
@freezed
class ProgressResponse with _$ProgressResponse {
  const factory ProgressResponse({
    required int solvedCount,
    required double correctRate,
    required int streakDays,
    ReadinessResponse? readiness,
  }) = _ProgressResponse;

  factory ProgressResponse.fromJson(Map<String, dynamic> json) =>
      _$ProgressResponseFromJson(json);
}
